File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int total=(1<<nums.size());
        vector<vector<int>>result;
        for(int i=0;i<total;i++)
        {
            vector<int>curr;
            for(int j=0;j<nums.size();j++)
            {
                if((1<<j)&i)
                    curr.push_back(nums[j]);
            }
            result.push_back(curr);
        }
        return result;
    }
};

File: 0078-subsets/0078-subsets.java
class Solution {

    private void helper(int indx,int[] nums,List<Integer> intermediate, List<List<Integer>> res)
    {
        if(indx==nums.length)
        {
            res.add(new ArrayList<>(intermediate));
            return;
        }
        helper(indx+1,nums,intermediate,res);
        intermediate.add(nums[indx]);
        helper(indx+1,nums,intermediate,res);
        intermediate.remove(intermediate.size()-1);
    }

    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        helper(0,nums, new ArrayList<>(),res);
        return res;
    }
}